src/governance.ts

```diff
@@ -128,8 +134,54 @@ export class GovernanceCanister {
     return this.responseConverters.toListProposalsResponse(rawResponse);
   };
 
+  public createNeuron = async ({
+    stake,
+    principal,
+    ledgerCanister,
+  }: {
+    stake: ICP;
+    principal: Principal;
+    ledgerCanister: LedgerCanister;
+  }): Promise<void> => {
+    if (stake.toE8s() < E8S_PER_ICP) {
+      throw "Need a minimum of 1 ICP to stake a neuron";
+    }
+
+    const nonceBytes = new Uint8Array(randomBytes(8));
+    uint8ArrayToBigInt;
+    const nonce = uint8ArrayToBigInt(nonceBytes);
+    const toSubAccount = buildSubAccount(nonceBytes, principal);
+
+    // Note - the account identifier here is just an alias of string.  Something to change?
+    const accountIdentifierAsString = principalToAccountIdentifier(
+      this.canisterId,
+      toSubAccount
+    );
+    const accountIdentifier = AccountIdentifier.fromHex(
+      accountIdentifierAsString
+    );
+
+    // Send amount to the ledger.
+    await ledgerCanister.transfer({
+      memo: nonce,
+      amount: stake,
+      to: accountIdentifier,
+      // TODO: Support subaccounts
+    });
+
+    // Notify the governance of the transaction so that the neuron is created.
+    await this.claimOrRefreshNeuronFromAccount({
+      controller: principal,
+      memo: nonce,
+    });
+  };
+
   /**
    * Gets the NeuronID of a newly created neuron.
+   *
+   * COMMENT: This is also used to create a neuron after transfering.
+   * QUESTION: Does it mean that the transfer creates the neuron and this just grabs it?
+   * QUESTION: Should this be private and expose only `createNeuron`?
    */
   public claimOrRefreshNeuronFromAccount = async (
```

Comment:
Turns out that `claim_or_refresh_neuron_from_account` is deprecated in favour of `manage_neuron.claim_or_refresh`. See here - https://gitlab.com/dfinity-lab/public/ic/-/blob/a8ec2c603c494dc650a253e90513503cd7ec9a64/rs/nns/governance/canister/canister.rs#L384

Reply:
Do you mind if we integrate this version for now and we refactor later? This functionality is blocking all the neurons page.

I already created the issue for the refactor: https://dfinity.atlassian.net/browse/L2-314